rfd/0080-yubikey-piv.md

```diff
@@ -0,0 +1,272 @@
+ - Users with the `superadmin` role are not required to use yubikey login (superadmin roles are inherently unsafe, might as well own it)
+```yaml
+kind: cluster_auth_preference
+version: v2
+metadata:
+  name: cluster-auth-preference
+spec:
+  yubikey_login: 
+    required: true
+---
+kind: role
+version: v5
+metadata:
+  name: admin
+spec:
+  yubikey_login: 
+    pin_policy: always
+    touch_policy: always
+---
+kind: role
+version: v5
+metadata:
+  name: superadmin
+spec:
+  yubikey_login: 
+    required: false
+```
+
+Example 2:
+ - Users with `yubikey-pin` role must use yubikey login with `pin=always` and `touch=cached|always`
+ - Users with `yubikey-touch` role must use yubikey login with `pin=once|always` and `touch=always`
+ - Users with both the `yubikey-pin` and `yubikey-touch` roles must use yubikey login  with `pin=always` and `touch=always`
+ - Users with neither the `yubikey-pin` nor `yubikey-touch` roles are not required to use yubikey login
+```yaml
+kind: cluster_auth_preference
+version: v2
+metadata:
+  name: cluster-auth-preference
+spec:
+  yubikey_login: 
+    required: false
+    pin_policy: once
+    touch_policy: cached
+---
+kind: role
+version: v5
+metadata:
+  name: yubikey-pin
+spec:
+  yubikey_login: 
+    required: true
+    pin_policy: always
+---
+kind: role
+version: v5
+metadata:
+  name: yubikey-touch
+spec:
+  yubikey_login: 
+    required: true
+    pin_policy: never
+```
+
+### UX
+
+#### yubikey slots
+
+Yubikey 5 has 5 different [key slots](https://docs.yubico.com/hardware/yubikey/yk-5/tech-manual/yk5-apps.html#slot-information) to choose from, one of which is reserved for attestation. `Slot 9a: PIV Authentication` is the most fitting option, and is the same slot used by `yubikey-agent`.
+
+If the slot is already in use, we will provide the user with a prompt to reuse this key, overwrite it, or abort. Users can also provide the `--yubikey-overwrite` or `--yubikey-reuse` flags to bypass this prompt.
+
+```
+> tsh login
+
+yubikey slot 9a has a pre-generated private key with public key:
+-----BEGIN PUBLIC KEY-----
+...
+-----END PUBLIC KEY-----
+
+Would you like to:
+ (1) Resuse this key
+ (2) Overwrite this key
+ (3) Abort login
+```
+
+In the case of a Teleport user connecting to multiple clusters, they'll need to use `--yubikey-reuse` in order to maintain access to them simultaneously.
+
+Note that reusing the key is not any less secure than generating a new one, as the Teleport Auth server can still attest that the key was generated on the yubikey, and therefore cannot be exported. 
+
+#### new `tsh` flags
+
+| Name | Default Value(s) | Allowed Value(s) | Description |
+| `-y, --yubikey` | - | - | During login, use yubikey to generate and store private key data for your login session. |
+| `--yubikey-pin-policy` | never | always once never | Can be provided with `tsh login -y` to control how often you need to enter your yubikey PIN to access the yubikey private key data. |
```

Comment:
Prefix all with `--yubikey-piv-`, in case we want to do similar settings for other yubi features?